### PR TITLE
Adds support for WASAPI provider and account

### LINKED ISSUE #67: Support WASAPI provider and collections
Each collection must have a WASAPI provider and an account. See https://github.com/sul-dlss/was-registrar-app/blob/master/config/settings/development.yml for an example of WASAPI providers / accounts.

The user should be able to select / update the provider and account on the create / edit page. To simplify the UI, it is OK if the selection of the two is combined (e.g., ait > ua, ait > history, notait > ua).

### PR DESCRIPTION
**Pull request by justinlittman**
closes #67

### CODE REVIEW

File: app/controllers/collections_controller.rb [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 # Controller for Collections
 class CollectionsController < ApplicationController
+  before_action :wasapi_provider_accounts
```
Comment (jcoyne) [inline app/controllers/collections_controller.rb]: I think this only needs to happen for :new and :edit, right?

File: app/models/collection.rb [hunk 2]
```diff
@@ -3,7 +3,15 @@
...
-  validates_presence_of :title, :druid, :active, :embargo_months
+  validates_presence_of :title, :druid, :active, :embargo_months, :wasapi_provider_account
   validates :druid, format: { with: /druid:.+/,
                               message: 'must begin with druid:' }
+
+  def wasapi_provider_account
+    wasapi_provider && wasapi_account ? "#{wasapi_provider}:#{wasapi_account}" : nil
+  end
+
+  def wasapi_provider_account=(wasapi_provider_account)
```
Comment (jcoyne) [inline app/models/collection.rb]: can you use a different variable name that isn't the same as a method?

File: app/models/collection.rb [hunk 3]
```diff
@@ -3,7 +3,15 @@
 # Web archive collection
 class Collection < ApplicationRecord
   self.primary_key = 'druid'
-  validates_presence_of :title, :druid, :active, :embargo_months
+  validates_presence_of :title, :druid, :active, :embargo_months, :wasapi_provider_account
```
Comment (jcoyne) [inline app/models/collection.rb]: Should this be validating `wasapi_provider` and `wasapi_account`?

Comment (justinlittman) [reply app/models/collection.rb]: At least from the UI perspective, only `wasapi_provider_account` is needed (not `wasapi_provider` and `wasapi_account` separately).

File: app/controllers/collections_controller.rb (hunk 1 above)
Comment (justinlittman) [reply app/controllers/collections_controller.rb]: Yes. Fixed.

File: app/models/collection.rb (hunk 2 above)
Comment (justinlittman) [reply app/models/collection.rb]: Done.

**Pull request merged**